sdk/core/core-client-rest/src/urlHelpers.ts

```diff
@@ -1,7 +1,38 @@
 // Copyright (c) Microsoft Corporation.
 // Licensed under the MIT License.
 
-import { RequestParameters } from "./common.js";
+import { PathParameterWithOptions, RequestParameters } from "./common.js";
+
+/**
+ * An object that can be passed as a query parameter, allowing for additional options to be set relating to how the parameter is encoded.
+ */
+interface QueryParameterWithOptions {
```

Comment:
As discussed, can we add `style` also with possible values: form | spaceDelimited | pipeDelimited?

Reply:
Added it, please take a look :)